opslevel/resource_opslevel_service_dependency.go

```diff
@@ -1,5 +1,209 @@
+	return &ServiceDependencyResource{}
+}
+
+// ServiceDependencyResource defines the resource implementation.
+type ServiceDependencyResource struct {
+	CommonResourceClient
+}
+
+// ServiceDependencyResourceModel describes the ServiceDependency managed resource.
+type ServiceDependencyResourceModel struct {
+	DependsUpon types.String `tfsdk:"depends_upon"`
+	Id          types.String `tfsdk:"id"`
+	Note        types.String `tfsdk:"note"`
+	Service     types.String `tfsdk:"service"`
+}
+
+func NewServiceDependencyResourceModel(ctx context.Context, serviceDependency opslevel.ServiceDependency) ServiceDependencyResourceModel {
+	return ServiceDependencyResourceModel{
+		Id:   ComputedStringValue(string(serviceDependency.Id)),
+		Note: OptionalStringValue(serviceDependency.Notes),
+	}
+}
+
+func (r *ServiceDependencyResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_serviceDependency"
+}
+
+func (r *ServiceDependencyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "ServiceDependency Resource",
+
+		Attributes: map[string]schema.Attribute{
+			"depends_upon": schema.StringAttribute{
+				Description: "The ID or alias of the service that is depended upon.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of the serviceDependency.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"note": schema.StringAttribute{
+				Description: "Notes for service dependency.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"service": schema.StringAttribute{
+				Description: "The ID or alias of the service with the dependency.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *ServiceDependencyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel ServiceDependencyResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serviceDependency, err := r.client.CreateServiceDependency(opslevel.ServiceDependencyCreateInput{
+		DependencyKey: opslevel.ServiceDependencyKey{
+			DestinationIdentifier: opslevel.NewIdentifier(planModel.DependsUpon.ValueString()),
+			SourceIdentifier:      opslevel.NewIdentifier(planModel.Service.ValueString()),
+		},
+		Notes: planModel.Note.ValueStringPointer(),
```

Comment:
I thought we should use

```suggestion
		Notes: opslevel.RefOf(planModel.Note.ValueString()),
```

Reply:
We do need the suggested change for update commands. Since this only creates new resources this is fine since it will be omitted. Changing this to RefOf might also explicitly set an empty string in the database instead of `null` maybe? But they amount to the same thing I think

Reply:
This works as is 👌 - I just created a resource using this config:
```tf
resource "opslevel_service_dependency" "example" {
  service      = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS8yMzc1"
  depends_upon = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS8xMTgyMTk"
}
```